cmake/std/atdm/utils/get_known_system_info.sh

```diff
@@ -31,6 +31,9 @@ fi
 source ${ATDM_CONFIG_SCRIPT_DIR}/utils/get_system_info_utils.sh
 
 realHostname=`hostname`
+if [[ "${ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE}" ]] ; then
```

Comment:
Please remove lines 34-36 or add a warning under the branch here, such as: `echo: "WARNING: realHostname=$realHostname overriden to value of ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE which is $ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE in $PWD/get_known_system_info.sh"`.

Reply:
@e10harvey, that is just to drive testing (initially manual testing but in the future, automated unit testing).  To make that clear, we could change the name to `ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE_FOR_UNIT_TESTING`?   And we should add a comment to describes the purpose of this variable as well.  Is that enough?

We can also add the echo "Warning" output too if you want.  But if someone sets a variable called `ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE_FOR_UNIT_TESTING` and is surprised about the behavior of this, I don't feel badly for them.

Reply:
Both the current variable name and ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE_FOR_UNIT_TESTING are OK. Documenting this variable would be excellent. Yes, please either add the warning or remove those lines.

Reply:
I resolved this in the commit 02303b7f0c.  I tested this on 'stria' with:

```
$ . cmake/std/atdm/load-env.sh default
Hostname 'stria-login1' matches known ATDM host 'stria-login1' and system 'van1-tx2'
Setting compiler and build options for build-name 'default'
Using ARM ATSE compiler stack ARM-20.0_OPENMPI-4.0.2 to build DEBUG code with Kokkos node type SERIAL

$ export ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE_FOR_UNIT_TESTING=stria-something

$ . cmake/std/atdm/load-env.sh default

***
*** WARNING: realHostname=stria-login1 overriden to value of
*** ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE_FOR_UNIT_TESTING='stria-something'
*** in <trilinos-dir>/cmake/std/atdm/utils/get_known_system_info.sh.
*** This variable should only be set for unit testing purposes!
***

Hostname 'stria-something' matches known ATDM host 'stria-something' and system 'van1-tx2'
Setting compiler and build options for build-name 'default'
Using ARM ATSE compiler stack ARM-20.0_OPENMPI-4.0.2 to build DEBUG code with Kokkos node type SERIAL

$ unset ATDM_CONFIG_GET_KNOW_SYSTEM_INFO_REAL_HOSTNAME_OVERRIDE_FOR_UNIT_TESTING

$ . cmake/std/atdm/load-env.sh default
Hostname 'stria-login1' matches known ATDM host 'stria-login1' and system 'van1-tx2'
Setting compiler and build options for build-name 'default'
Using ARM ATSE compiler stack ARM-20.0_OPENMPI-4.0.2 to build DEBUG code with Kokkos node type SERIAL
```

That looks correct to me.

NOTE: You can't use `$PWD/get_known_system_info.sh` because the current directory is not `<trilinos-dir>/cmake/std/atdm/utils` when this script is run.  I also decided not to put in the full path since it makes it really long.  People can figure out how to find that if they need to.